### PR TITLE
added logger to debug with contention id

### PR DESCRIPTION
**Pull request by harishvanga333**
Resolves #{github issue number}

### Description
Please explain the changes you made here.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Go to ...

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### Database Changes
*Only for Schema Changes*

* [ ] Add typical timestamps (`created_at`, `updated_at`) for new tables
* [ ] Update column comments; include a "PII" prefix to indicate definite or potential [PII data content](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii)
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`) (see [Writing DB migrations](https://github.com/department-of-veterans-affairs/caseflow/wiki/Writing-DB-migrations))
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Perform query profiling (eyeball Rails log, check bullet and fasterer output)
* [ ] Add appropriate indexes (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] Run `make check-fks`; add any missing foreign keys or add to `config/initializers/immigrant.rb` (see [Record associations and Foreign Keys](https://github.com/department-of-veterans-affairs/caseflow/wiki/Record-associations-and-Foreign-Keys))
* [ ] Add `belongs_to` for associations to enable the [schema diagrams](https://department-of-veterans-affairs.github.io/caseflow/task_trees/schema/schema_diagrams) to be automatically updated
* [ ] Post this PR in #appeals-schema with a summary
* [ ] Document any non-obvious semantics or logic useful for interpreting database data at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

### Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
* [ ] Update Fakes
* [ ] Integrations impacting functionality are tested in Caseflow UAT

### CODE REVIEW

File: app/models/request_issue_contention.rb [hunk 1]
```diff
@@ -16,8 +16,14 @@ def vbms_contention
 
   def update_text!
     return unless contention_reference_id && contention_text_update_pending
-
-    fail EndProductEstablishment::ContentionNotFound, contention_reference_id unless vbms_contention
+    # if Vbms_contention is nil, then it will give the ContentionNotFound error message
+    unless vbms_contention
+      Raven.capture_exception("ContentionNotFound: Contention reference id #{contention_reference_id} not found while updating request issue.")
+      self.error_id = Raven.last_event_id || "000000000000000001234567890abcdefg"
```
Comment (nkutub) [inline app/models/request_issue_contention.rb:22]: Can you explain this line? why are you assigning this random value if last_ervent_id is nil ? will this happen?

Comment (nkutub) [inline app/models/request_issue_contention.rb:22]: I don't see where error_id is declared? why is this set ? this is not active record and it does not subclass anything else. You are also setting an attribute but there is not save , so how is this being used.

Comment (harishvanga333) [reply app/models/request_issue_contention.rb:22]: i am following same pattern what we using in different files app/controllers/intakes_controller.rb

Comment (harishvanga333) [reply app/models/request_issue_contention.rb:22]: in log we added the contention_id, thats the id is required

**Comment by nkutub** [comment]
@harishvanga333 , I left comments also I noticed that you have some rubocop issues on clodeclimate. I know that other CI errors are not related to your PR , but we should not add to mix. 

Also I would like to see an rspec with some doubles added since this is a very small class and your change is small but I see some potential errors. I would like to see a spec that runs the code you added and has expected calls to the dependencies.

File: app/models/request_issue_contention.rb (hunk 1 above)
Comment (nkutub) [reply app/models/request_issue_contention.rb:22]: yes I noticed that but there are many calls to ravin in other parts of the code that do not do this.

Comment (nkutub) [reply app/models/request_issue_contention.rb:22]: this attriute is only defined on the IntakesController, which this class is not associated with how will this attribute be declared

**Review by nkutub — APPROVED** [review]

**Pull request closed without merging**